README.md
# STAF
Forked from the Software Testing Automation Framework (STAF), which is a framework designed to improve the level of reuse and automation in test cases and test environments. The goal of STAF is to provide a complete end-to-end automation solution for testers.
# Build
###Core
```
export PROJECTS="staf connprov*"
export OS_NAME=win32 (See above for other valid OS_NAME values)
export BUILD_TYPE=retail
export MSVCDIR=d:/MVStudio (Only needed if on Windows)
export VSCOMMONDIR=d:/MVStudio/Common (Only needed if on Windows)
export MSSDKDIR=d:/MicrosoftSDK (Only needed if on Windows and building IA-64 code)
export STAF_USE_IPV6= (Make sure not set to build TCP IPv4 libraries)
export OPENSSL_ROOT=c:/tools/openssl-0.9.8e-root (Needed if STAF_USE_SSL is set to 1)
make
```
###Java
```
export PROJECTS=java
export JAVA_DRIVE=d
export JAVA_LIBS=jvm
export JAVA_VERSION=1.4
export JAVA_BUILD_V11=0
export JAVA_BUILD_V12=1
export JAVA_V12_ROOT=/cygdrive/d/ibmjdk1.4.2
export JAVA_V12_LIBDIRS=/cygdrive/d/ibmjdk1.4.2/lib
export JAVA_V12_BIN_DIR=/usr/java_dev2/sh (this is only required on AIX, when using Java 1.2+ to build)
make
```
###Package
```
export OS_NAME=linux
make PROJECTS=pkg
```

History
-------------------------------------------------------------------------------
History Log for STAF  
  
  Legend:
   - Fixes
   + Features
   = Internal changes
   * Changes required to migrate from one release to another

  Notes:
  1) To get more information on each bug, use one of the following urls,
     replacing Ticket# with the Bug#:
       https://sourceforge.net/p/staf/bugs/Ticket#/
     or use the old format for bugs created before June 2013:
       https://sourceforge.net/support/tracker.php?aid=Ticket#  
     Examples:
       https://sourceforge.net/p/staf/bugs/1484/
       https://sourceforge.net/support/tracker.php?aid=3608059
  2) To get more information on each feature, use one of the following urls,
     replacing Ticket# with the Feature#:
       https://sourceforge.net/p/staf/feature-requests/Ticket#/
     or use the old format for features created before June 2013:
       https://sourceforge.net/support/tracker.php?aid=Ticket#
     Examples:
       https://sourceforge.net/p/staf/feature-requests/723/
       https://sourceforge.net/support/tracker.php?aid=3606436
    
--------------------------------------------------------------------------------

Version 3.4.26: 12/30/2016

  - Fixed a problem where starting STAFProc on AIX failed with a core dump
    (Bug #1585)
  - Fixed a problem where STAFProc may fail to start if a STAF Java service
    registration takes a long time.  Also, added logging a message in the
    JVM Log with the number of times it attempted to connect to the JVM
    (Bug #1586)
  - Updated the error message for return code 51 (Directory Copy Error) to say
    more details about why a specific file could not be copied may be provided
    by submitting a FS COPY FILE request (Bug #1588)
  + Upgraded the STAF IA installers to use InstallAnywhere 2015 SP1 Public
    Hotfix A for InstallAnywhere (equivalent to IBM Hotfix A-D) (Feature #779)
  + Upgraded the IA 2015 SP1 to to use InstallAnywhere Public Hotfix for
    Windows Launcher (Dec 2016) which contains security fixes (Feature #780)
  
--------------------------------------------------------------------------------

Version 3.4.25: 06/28/2016

  + Changed the STAF Windows build system and InstallAnywhere build system from
    Windows Server 2008 x86 to Windows Server 2008 R2 x86_64 (Feature #773)
  + Upgraded the STAF IA installers to use InstallAnywhere 2015 Premier Edition
    Service Pack 1 (SP1) with IBM Hotfix A (Feature #776)
  - Fixed a typo in the description of the DIRECTORY option for the ZIP
    service's ADD request in the STAF User's Guide (Bug #1572)
  - Updated the help text for RC 21 and the STAF FAQ with a note about when
    STAFProc is run as an Administrator on Windows with UAC enabled, RC 21 is
    returned if a STAF service request is not also run as an Administrator
    (Bug #1573)
  + Upgraded the STAF IA installers to use IBM HotFix B for InstallAnywhere
    2015 SP1 which contains a fix for a Solaris installer bug (Feature #777)
  - Fixed problems in the BuildSTAF.xml STAX job when a build fails to make
    sure an email is sent, a testcase fail is posted, and to not try to
    install/test the bad build (Bug #1574)
  + Changed the ssl interface to use the TLS V1.2 protocol (which required
    building STAF using OpenSSL 1.0.2g instead of 0.9.8e) and disabling the
    SSLv2, SSLv3, TLSv1, and TLSv1.1 protocols to fix a "TLS V1.2 Protocol
    Detection" TCP/IP Vulnerability (Feature #774) 
  - Reissued the default certificate for the SSL network interface with a
    stonger hashing algorithm (SHA-256) to fix TCP/IP vulnerability scan issue
    "SSL Certificate Signed Using Weak Hashing Algorithm". All STAF systems
    that communicate using the ssl interface with the default certificate
    should be upgraded to V3.4.25+ so that they use the same certificate in
    order to avoid RC 16 with error message "STAFConnectionProviderConnect:
    Error in client SSL handshake" (Bug #1575)
  - Fixed many compiler warnings seen when building STAF on Mac OS X 10.10 and
    other operating systems (Bug #1576)
  - Changed sMaxLineLength's type from int to unsigned int in STAFConfigService
    to resolve warnings when compiling STAF on Windows 64-bit and other 64-bit
    operating systems (Bug #1579)
    
--------------------------------------------------------------------------------

Version 3.4.24.1: 01/06/2015

  - Fixed a problem for Windows 8.1, Windows Server 2012 R2, Windows 10, and
    later where STAF was incorrectly setting the Windows version to 6.2 and
    incorrectly setting the STAF/Config/OS/Name variable to Win8
    (Bugs #1568 and #1569)
    
--------------------------------------------------------------------------------

Version 3.4.24: 12/15/2015

  - Updated section "4.4.2 JSTAF service proxy library" in the STAF User's Guide
    to state that STAF Java support on Mac OS X requires Java 8.0 or newer
    (Bug #1560)
  - Reissued the default certificate for the SSL network interface with a longer
    RSA key length (2048 bits instead of 1024) to fix TCP/IP vulnerability

webpages/sf/index.php
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>

<head>
        <title>Software Testing Automation Framework (STAF)</title>
</head>

<body>

<?php
require "top.php";
require "navigate.php";
?>

<!-- Insert text for home page here -->

<tr>
<td>
<font face="arial, helvetica" color="#1e029a" size="4"><b>Welcome to STAF</b></font>
<p>
The Software Testing Automation Framework (STAF) is an open source, multi-platform,
multi-language framework designed around the idea of reusable components, called services
(such as process invocation, resource management, logging, and monitoring).
STAF removes the tedium of building an automation infrastructure, thus enabling you to
focus on building your automation solution.  The STAF framework provides the foundation
upon which to build higher level solutions, and provides a pluggable approach supported
across a large variety of platforms and languages.
<p>
<font face="arial, helvetica" color="#1e029a" size="4"><b>Latest Releases</b></font>
<p>
<ul compact>
<li><b>STAF 3.4.25</b> (June 28, 2016) (<a href="getcurrent.php">Download</a>) 
<li><b>STAX 3.5.17</b> (June 28, 2016) (<a href="getstax.php">Download</a>)
</ul>
<p>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
<b>Other STAF Services</b> (provided separately as "add-ons" to the core STAF services):
<menu>
<li><b>Cron Service 3.4.0</b> (June 30, 2011) (<a href="getservices.php#Cron">Download</a>)
<li><b>Email Service 3.3.9</b> (June 26, 2015) (<a href="getservices.php#Email">Download</a>)
<li><b>Event Service 3.1.5</b> (September 28, 2012) (<a href="getservices.php#Event">Download</a>)
<li><b>EventManager Service 3.4.0</b> (June 30, 2011) (<a href="getservices.php#EventManager">Download</a>)
<li><b>FSExt (File System Extension) Service 3.0.2</b> (December 15, 2009) (<a href="getservices.php#FSExt">Download</a>)
<li><b>FTP Service 1.0.3</b> (December 7, 2010) (<a href="getservices.php#FTP">Download</a>)
<li><b>HTTP Service 3.0.4</b> (December 9, 2011) (<a href="getservices.php#HTTP">Download</a>)
<li><b>NamedCounter Service 3.0.2</b> (December 15, 2009) (<a href="getservices.php#NamedCounter">Download</a>)
<li><b>Namespace Service 1.0.3</b> (September 28, 2011) (<a href="getservices.php#Namespace">Download</a>)
<li><b>SXE (Staf eXecution Engine) Service 3.0.3</b> (December 15, 2009) (<a href="getservices.php#SXE">Download</a>)
<li><b>Timer Service 3.0.3</b> (December 15, 2009) (<a href="getservices.php#Timer">Download</a>)
</menu>
<p>
STAF can be leveraged to help solve common industry problems, such as more frequent
product cycles, less preparation time, reduced testing time, more platform choices,
more programming language choices, and increased National Language requirements.
STAF can help in these areas since it is a proven and mature technology, promotes
automation and reuse, has broad platform and language support, and provides a common
infrastructure across teams.
<p>
STAX is an execution engine which can help you thoroughly automate the
distribution, execution, and results analysis of your testcases. STAX builds on top of
three existing technologies, STAF, XML, and Python, to place great automation power in
the hands of testers. STAX also provides a powerful monitoring application which
allows you to interact with and monitor the progress of your jobs.
Some of the main features of STAX are:  support for parallel execution, user-defined
granularity of execution control, support for nested testcases, the ability to control
the length of execution time, the ability to import modules at run-time, support for
existing Python and Java modules and packages, and the ability to extend both the
STAX language as well as the STAX monitoring application.
Using these capabilities, you can build sophisticated scripts to automate your
entire test environment, while ensuring maximum efficiency and control.
<p>
Other STAF services are also provided to help you to create an end-to-end automation
solution.  By using these services in your test cases and automated solutions,
you can develop more robust, dynamic test cases and test environments.
<p>
We encourage the community to post patches, new features, new STAF services, etc.
that enhance or extend the STAF project.  Please see <a href="contributions.php">
Making Contributions to STAF</a> for more information on how to do this.
<p>
Also, feel free to post a question via one of our
<a href="http://sourceforge.net/p/staf/discussion/">discussion forums</a>
or <a href="mailinglists.php">mailing lists</a>.  These are also good places
to search for answers to questions, in addition to the <a href="current/STAFFAQ.htm">FAQ</a>,
<a href="current/STAFUG.htm">STAF User Guide</a>, <a href="current/STAX/staxug.html">STAX User Guide</a>,
and <a href="docs.php">other documentation</a>.
<p>
Note that eWeek.com ran an article in 2007 about the most important open-source applications.
At the end they invited readers to write in letting them know about applications
they missed.  Then they ran an article containing the ones most often mentioned
and their readers said STAF was one of the most important open-source applications.
See the full article
<a href="http://www.eweek.com/slideshow_viewer/0,1205,l=&s=25947&a=209454&po=1,00.asp?p=y?kc=EWKNLNAV061307FEA1">
here</a>.  This <a href="http://www.eweek.com/slideshow_viewer/0,1205,l=&s=25947&a=209454&po=13,00.asp">slide</a>
is about STAF.
<p>
Also, SDTimes.com ran an article on July 22, 2008 that put STAF at the top of its
list of open source tools for Automated Testing.  See the
<a href="http://www.sdtimes.com/content/article.aspx?ArticleID=32583">
Functional Testing Tools, the Open-Source Wave</a> article.
<p>
<font face="arial, helvetica" color="#1e029a" size="4"><b>Latest News</b></font><BR>
<ul>
  <li><b>June 28, 2016:</b> <a href="getcurrent.php">STAF 3.4.25</a> and <a href="getstax.php">STAX 3.5.17</a>
       are now available.
  <li><b>January 7, 2016:</b> <a href="getcurrent.php">STAF 3.4.24.1</a> (updated only for Windows 32-bit and 64-bit) 
       is now available.
  <li><b>December 15, 2015:</b> <a href="getcurrent.php">STAF 3.4.24</a> and <a href="getstax.php">STAX 3.5.16</a>
       are now available.
  <li><b>September 29, 2015:</b> <a href="getcurrent.php">STAF 3.4.23</a> and <a href="getstax.php">STAX 3.5.15</a>
       are now available.
  <li><b>June 26, 2015:</b> <a href="getcurrent.php">STAF 3.4.22</a>, <a href="getstax.php">STAX 3.5.14</a>,
       and <a href="getservices.php#Email">Email 3.3.9</a> are now available.
  <li><b>March 31, 2015:</b> <a href="getstax.php">STAX 3.5.13</a> is now available.
  <li><b>March 30, 2015:</b> <a href="getcurrent.php">STAF 3.4.21</a> is now available.
  <li><b>December 10, 2014:</b> <a href="getstax.php">STAX 3.5.12</a> is now available.
  <li><b>November 7, 2014:</b> <a href="getcurrent.php">STAF 3.4.20</a> is now available.
  <li><b>September 24, 2014:</b> <a href="getcurrent.php">STAF 3.4.19</a> and <a href="getstax.php">STAX 3.5.11</a>,
       are now available.
  <li><b>June 30, 2014:</b> <a href="getcurrent.php">STAF 3.4.18</a> and <a href="getstax.php">STAX 3.5.10</a>,
       are now available.
  <li><b>March 30, 2014:</b> <a href="getcurrent.php">STAF 3.4.17</a> and <a href="getstax.php">STAX 3.5.9</a>,
       are now available.
  <li><b>December 16, 2013:</b> <a href="getcurrent.php">STAF 3.4.16</a> and <a href="getstax.php">STAX 3.5.8</a>,
       are now available.
  <li><b>September 30, 2013:</b> <a href="getcurrent.php">STAF 3.4.15</a> and <a href="getstax.php">STAX 3.5.7</a>,
       are now available.
  <li><b>June 28, 2013:</b> <a href="getcurrent.php">STAF 3.4.14</a>, <a href="getstax.php">STAX 3.5.6</a>,
       and <a href="getservices.php">Email 3.3.8</a> are now available.
  <li><b>March 29, 2013:</b> <a href="getcurrent.php">STAF 3.4.13</a>, <a href="getstax.php">STAX 3.5.5</a>,
       and <a href="getservices.php">Email 3.3.7</a> are now available.
  <li><b>December 14, 2012:</b> <a href="getcurrent.php">STAF 3.4.12</a> is now available.
  <li><b>September 28, 2012:</b> <a href="getcurrent.php">STAF 3.4.11</a>, <a href="getstax.php">STAX 3.5.4</a>,
       and <a href="getservices.php">Event 3.1.5</a> are now available.
  <li><b>June 29, 2012:</b> <a href="getcurrent.php">STAF 3.4.10</a> is now available.
  <li><b>March 29, 2012:</b> <a href="getcurrent.php">STAF 3.4.9</a> and <a href="getstax.php">STAX 3.5.3</a>
       are now available.
  <li><b>December 9, 2011:</b> <a href="getcurrent.php">STAF 3.4.8</a>, <a href="getstax.php">STAX 3.5.2</a>,
       and <a href="getservices.php">HTTP 3.0.4</a> are now available.
  <li><b>September 28, 2011:</b> <a href="getcurrent.php">STAF 3.4.7</a>, <a href="getstax.php">STAX 3.5.1</a>,
       <a href="getservices.php">Email 3.3.6, HTTP 3.0.3, and Namespace 1.0.3</a> are now available.
  <li><b>June 30, 2011:</b> <a href="getcurrent.php">STAF 3.4.6</a>, <a href="getstax.php">STAX 3.5.0</a>,
       <a href="getservices.php">Cron 3.4.0, and EventManager 3.4.0</a> are now available.
  <li><b>March 31, 2011:</b> <a href="getcurrent.php">STAF 3.4.5</a> is now available. 
  <li><b>December 14, 2010:</b> <a href="getcurrent.php">STAF 3.4.4</a> and
         <a href="getstax.php">STAX 3.4.5</a> are now available. 
  <li><b>December 7, 2010:</b> <a href="getservices.php#FTP">FTP 1.0.3</a> is now available.
  <li><b>December 6, 2010:</b> <a href="getservices.php#Email">Email 3.3.5</a> is now available.
  <li><b>December 2, 2010:</b> <a href="getstax.php">STAX 3.5.0 Beta 1</a> is now available.
  <li><b>November 1, 2010:</b> <a href="getservices.php">Cron 3.3.8 and EventManager 3.3.8</a> are now available.
  <li><b>September 30, 2010:</b> <a href="getstax.php">STAX 3.4.4</a>, <a href="getservices.php">Cron 3.3.7,
         and EventManager 3.3.7</a> are now available.
  <li><b>September 28, 2010:</b> <a href="getcurrent.php">STAF 3.4.3</a> is now available.
  <li><b>August 24, 2010:</b> <a href="getservices.php#Email">Email 3.3.4</a> is now available.
  <li><b>June 29, 2010:</b> <a href="getcurrent.php">STAF 3.4.2</a> and
         <a href="getstax.php">STAX 3.4.3</a> are now available.
  <li><b>May 20, 2010:</b> <a href="education.php">STAF/STAX Education V3.4.1</a> is now available.
  <!--
  <li><b>May 10, 2010:</b> <a href="getstax.php">STAX 3.4.2</a> is now available.
  <li><b>March 30, 2010:</b> <a href="getcurrent.php">STAF 3.4.1</a> and
         <a href="getstax.php">STAX 3.4.1</a> are now available. 
  <li><b>December 18, 2009:</b> <a href="getcurrent.php">STAX 3.4.0</a> is now available. 
  <li><b>December 15, 2009:</b> <a href="getservices.php">Cron 3.3.6, Email 3.3.3,
         Event 3.1.4, EventManager 3.3.6, FSExt 3.0.2, FTP 1.0.2, HTTP 3.0.2,
         NamedCounter 3.0.2, Namespace 1.0.2, SXE 3.0.3, and Timer 3.0.3</a> are now available.
  <li><b>December 14, 2009:</b> <a href="getcurrent.php">STAF 3.4.0</a> is now available. 

  <li><b>September 30, 2009:</b> <a href="getcurrent.php">STAF 3.3.5</a> and
         <a href="getstax.php">STAX 3.3.8</a> are now available. 
  <li><b>July 13, 2009:</b> <a href="getcurrent.php">STAF 3.3.4.1</a> (with an important bug fix for Unix)
         is now available. 
  <li><b>June 30, 2009:</b> <a href="getcurrent.php">STAF 3.3.4</a> and
         <a href="getstax.php">STAX 3.3.7</a> are now available. 
  <li><b>May 29, 2009:</b> <a href="getservices.php">Event 3.1.3</a> is now available.
  <li><b>April 29, 2009:</b> <a href="getservices.php">Cron 3.3.5 and EventManager 3.3.5</a> are now available.
  <li><b>March 31, 2009:</b> <a href="getcurrent.php">STAF 3.3.3</a> and
         <a href="getstax.php">STAX 3.3.6</a> are now available. 
  <li><b>December 10, 2008:</b> <a href="getservices.php">Cron 3.3.4 and EventManager 3.3.4</a> are now available.
  <li><b>December 8, 2008:</b> <a href="getcurrent.php">STAF 3.3.2</a> and
         <a href="getstax.php">STAX 3.3.5</a> are now available. 
  <li><b>October 6, 2008:</b> <a href="getservices.php">FTP 1.0.1</a> is now available.
  <li><b>September 24, 2008:</b> <a href="getcurrent.php">STAF 3.3.1</a>,
                                <a href="getstax.php">STAX 3.3.4</a>, 
                                <a href="getservices.php">Cron 3.3.3, Email 3.3.2, Event 3.1.2, 
                                EventManager 3.3.3, and HTTP 3.0.1</a> are now available.
  <li><b>June 27, 2008:</b> <a href="getcurrent.php">STAF 3.3.0</a> and <a href="getstax.php">STAX 3.3.3</a>
         are now available.
  <li><b>June 20, 2008:</b> <a href="getservices.php">HTTP 3.0.0</a> is now available.
  <li><b>February 27, 2008:</b> <a href="getcurrent.php">STAF 3.2.5</a>,
                                <a href="getstax.php">STAX 3.3.2</a>, 
                                <a href="getservices.php">Cron 3.3.2, Email 3.3.1, Event 3.1.1, 
                                EventManager 3.3.2, FSExt 3.0.1, NamedCounter 3.0.1, Namespace 1.0.1,
                                SXE 3.0.2, and Timer 3.0.2</a> are now available.
  <li><b>January 8, 2008:</b> <a href="getservices.php">Cron 3.3.1 and EventManager 3.3.1</a> are now available.
  <li><b>December 13, 2007:</b> <a href="getstax.php">STAX 3.3.1</a> is now available.
  <li><b>December 13, 2007:</b> <a href="getcurrent.php">STAF 3.2.4</a> is now available.
  <li><b>November 20, 2007:</b> <a href="getservices.php">Cron 3.3.0 and EventManager 3.3.0</a> are now available.
  <li><b>November 9, 2007:</b> <a href="getservices.php">Email 3.3.0</a> is now available.
  <li><b>October 2, 2007:</b> <a href="getstax.php">STAX 3.3.0</a> is now available.
  <li><b>August 29, 2007:</b> <a href="getcurrent.php">STAF 3.2.3</a> is now available.
  <li><b>May 21, 2007:</b> <a href="getstax.php">STAX 3.2.1</a> is now available.
  <li><b>May 21, 2007:</b> <a href="getcurrent.php">STAF 3.2.2</a> is now available.
  <li><b>February 28, 2007:</b> <a href="getstax.php">STAX 3.2.0</a> is now available.
  <li><b>February 28, 2007:</b> <a href="getcurrent.php">STAF 3.2.1</a> is now available.
  <li><b>November 28, 2006:</b> <a href="getservices.php">Email 3.2.0</a> is now available.
  <li><b>November 9, 2006:</b> <a href="getstax.php">STAX 3.1.5</a> is now available.
  <li><b>October 31, 2006:</b> <a href="getcurrent.php">STAF 3.2.0</a> is now available.
  <li><b>August 31, 2006:</b> <a href="getservices.php">Cron 3.2.0 and EventManager 3.2.0</a> are now available.
  <li><b>August 29, 2006:</b> <a href="getcurrent.php">STAF 3.1.5</a> is now available.
  <li><b>July 12, 2006:</b> <a href="getstax.php">STAX 3.1.4</a> is now available.
  <li><b>June 29, 2006:</b> <a href="getstax.php">STAX 3.1.3</a> is now available.
  <li><b>June 29, 2006:</b> <a href="getcurrent.php">STAF 3.1.4</a> is now available.
  <li><b>May 18, 2006:</b> <a href="getstax.php">STAX 3.1.2</a> is now available.
 -->
  <li><b>April 21, 2006:</b> <a href="getcurrent2x.php">End of service for all STAF V2 releases.</a>
<!--
  <li><b>April 19, 2006:</b> <a href="getcurrent2x.php">STAF 2.6.11</a> is now available.
  <li><b>April 19, 2006:</b> <a href="getstax2x.php">STAX 1.5.10</a> is now available.
  <li><b>March 28, 2006:</b> <a href="getcurrent.php">STAF 3.1.3</a> is now available.
  <li><b>March 22, 2006:</b> End of service for all STAF V2 releases is planned for April 21, 2006.
  <li><b>January 16, 2006:</b> <a href="getcurrent.php">STAF 3.1.2</a> is now available.
  <li><b>December 21, 2005:</b> <a href="getcurrent2x.php">STAF 2.6.10</a> is now available.
  <li><b>December 9, 2005:</b> <a href="getcurrent.php">STAF 3.1.1</a> is now available.
  <li><b>December 8, 2005:</b> <a href="getstax.php">STAX 3.1.1</a> is now available.
  <li><b>December 8, 2005:</b> <a href="getstax2x.php">STAX V1.5.9</a> is now available.
  <li><b>October 4, 2005:</b> <a href="getstax.php">STAX 3.1.0</a> is now available.
  <li><b>September 30, 2005:</b> <a href="getcurrent.php">STAF 3.1.0</a> is now available.
  <li><b>August 17, 2005:</b> <a href="getcurrent2x.php">STAF 2.6.9</a> is now available.
  <li><b>August 17, 2005:</b> <a href="getstax2x.php">STAX 1.5.8</a> is now available.
  <li><b>August 16, 2005:</b> <a href="getcurrent.php">STAF 3.0.3</a> is now available.
  <li><b>August 16, 2005:</b> <a href="getstax.php">STAX 3.0.2</a> is now available.
  <li><b>July 19, 2005:</b> <a href="getcurrent.php">STAF 3.0.2</a> is now available.
  <li><b>June 28, 2005:</b> <a href="getcurrent.php">STAF 3.0.1</a> is now available.
  <li><b>June 28, 2005:</b> <a href="getstax.php">STAX 3.0.1</a> is now available.
  <li><b>May 5, 2005:</b> <a href="getcurrent2x.php">STAF 2.6.8</a> is now available.
  <li><b>May 5, 2005:</b> <a href="getstax2x.php">STAX 1.5.7</a> is now available.
  <li><b>April 22, 2005:</b> <a href="getcurrent.php">STAF 3.0.0</a> is now available.
  <li><b>April 22, 2005:</b> <a href="getstax.php">STAX 3.0.0</a> is now available.
  <li><b>January 26, 2005:</b> <a href="getcurrent2x.php">STAF 2.6.7</a> is now available.
  <li><b>January 26, 2005:</b> <a href="getstax2x.php">STAX 1.5.6</a> is now available.
  <li><b>December 14, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 7</a> is now available.
  <li><b>November 19, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 6</a> is now available.
  <li><b>November 5, 2004:</b> <a href="getstax2x.php">STAX 1.5.5</a> is now available.
  <li><b>November 5, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.6</a> is now available.
  <li><b>October 31, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 5</a> is now available.
  <li><b>September 30, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 4</a> is now available.
  <li><b>September 14, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.5</a> is now available.
  <li><b>August 2, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.4</a> is now available.
  <li><b>July 22, 2004:</b> <a href="getstax2x.php">STAX 1.5.4</a> is now available.
  <li><b>June 30, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 3</a> is now available.
  <li><b>May 26, 2004:</b> <a href="getstax2x.php">STAX 1.5.3</a> is now available.
  <li><b>May 14, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.3</a> is now available.
  <li><b>May 3, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 2</a> is now available.
  <li><b>April 30, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.2</a> is now available.
  <li><b>April 2, 2004:</b> <a href="getcurrent.php">STAF 3.0.0 Beta 1</a> is now available.
  <li><b>April 2, 2004:</b> <a href="getstax2x.php">STAX 1.5.2</a> is now available.
  <li><b>April 2, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.1</a> is now available.
  <li><b>March 4, 2004:</b> <a href="getstax2x.php">STAX 1.5.1</a> is now available.
  <li><b>March 4, 2004:</b> <a href="getcurrent2x.php">STAF 2.6.0</a> is now available.
  <li><b>January 29, 2004:</b> <a href="getstax2x.php">STAX 1.5.0</a> is now available.
  <li><b>January 29, 2004:</b> <a href="getcurrent2x.php">STAF 2.5.2</a> is now available.
  <li><b>September 30, 2003:</b> <a href="getstax.php">STAX 1.4.1</a> is now available.
  <li><b>September 29, 2003:</b> <a href="getcurrent.php">STAF 2.5.1</a> is now available.
  <li><b>July 30, 2003:</b> <a href="getcurrent.php">STAF 2.5.0</a> is now available.
  <li><b>June 4, 2003:</b> <a href="getstax.php">STAX 1.4.0</a> is now available.
  <li><b>March 27, 2003:</b> <a href="getcurrent.php">STAF 2.4.5</a> is now available.
  <li><b>March 18, 2003:</b> <a href="getstax.php">STAX 1.3.3</a> is now available.
  <li><b>March 13, 2003:</b> <a href="getcurrent.php">STAF 2.4.4</a> is now available.
  <li><b>December 11, 2002:</b> <a href="getstax.php">STAX 1.3.2</a> is now available.
  <li><b>December 10, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.4.3/">STAF 2.4.3</a> is now available.
  <li><b>October 8, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.4.2/">STAF 2.4.2 and STAX 1.3.1</a> are now available.
  <li><b>August 29, 2002:</b> <a href="getstax.php">STAX 1.2.1</a> is now available.
  <li><b>August 27, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.4.1/">STAF 2.4.1</a> is now available.
  <li><b>August 5, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.4.0/">STAF 2.4.0</a> is now available.
  <li><b>June 3, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.3.2/">STAF 2.3.2</a> is now available for Windows, Linux, AIX, and Solaris.
  <li><b>April 11, 2002:</b> <a href="http://sourceforge.net/projects/staf/files/staf/V2.3.1/">STAF 2.3.1</a> is now available for Windows and Linux.
  -->  
</ul>

</td>
</tr>

<!-- End of text for home page -->

<?php
require "bottom.php";
?>

</body>
</html>

